subgraphs/hop-protocol/protocols/hop-protocol/src/L2/amm.ts

```diff
@@ -0,0 +1,356 @@
+import { SDK } from "../../../../src/sdk/protocols/bridge";
+import { TokenPricer } from "../../../../src/sdk/protocols/config";
+import {
+  TokenInitializer,
+  TokenParams,
+} from "../../../../src/sdk/protocols/bridge/tokens";
+import {
+  BridgePermissionType,
+  BridgePoolType,
+} from "../../../../src/sdk/protocols/bridge/enums";
+import { BridgeConfig } from "../../../../src/sdk/protocols/bridge/config";
+import { Versions } from "../../../../src/versions";
+import { NetworkConfigs } from "../../../../configurations/configure";
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  log,
+  dataSource,
+  Bytes,
+} from "@graphprotocol/graph-ts";
+import {
+  TokenSwap,
+  AddLiquidity,
+  RemoveLiquidity,
+  RemoveLiquidityOne,
+} from "../../../../generated/HopL2Amm/L2_Amm";
+import { Token } from "../../../../generated/schema";
+import { getUsdPricePerToken, getUsdPrice } from "../../../../src/prices/index";
+import { bigIntToBigDecimal } from "../../../../src/sdk/util/numbers";
+import {
+  BIGINT_TEN_TO_EIGHTEENTH,
+  SIX,
+} from "../../../../src/sdk/util/constants";
+import { updateAMMTVE } from "../../../../src/sdk/util/tokens";
+
+class Pricer implements TokenPricer {
+  getTokenPrice(token: Token): BigDecimal {
+    const price = getUsdPricePerToken(Address.fromBytes(token.id));
+    return price.usdPrice;
+  }
+
+  getAmountValueUSD(token: Token, amount: BigInt): BigDecimal {
+    const _amount = bigIntToBigDecimal(amount, token.decimals);
+    return getUsdPrice(Address.fromBytes(token.id), _amount);
+  }
+}
+
+const conf = new BridgeConfig(
+  "0x03d7f750777ec48d39d080b020d83eb2cb4e3547",
+  "HOP-"
+    .concat(dataSource.network().toUpperCase().replace("-", "_"))
+    .concat("-BRIDGE"),
+  "hop-".concat(dataSource.network().replace("-", "_")).concat("-bridge"),
+  BridgePermissionType.PERMISSIONLESS,
+  Versions
+);
+
+class TokenInit implements TokenInitializer {
+  getTokenParams(address: Address): TokenParams {
+    const tokenConfig = NetworkConfigs.getTokenDetails(address.toHex());
+    const name = tokenConfig[1];
+    const symbol = tokenConfig[0];
+    const decimals = BigInt.fromString(tokenConfig[2]).toI32();
+    return { name, symbol, decimals };
+  }
+}
+
+export function handleTokenSwap(event: TokenSwap): void {
+  if (NetworkConfigs.getPoolsList().includes(event.address.toHexString())) {
+    const amount = event.params.tokensSold;
+
+    const fees = amount
+      .times(BigInt.fromString("4"))
+      .div(BigInt.fromString("10000"));
+    log.warning("FEES 2- fees: {}, fees: {}, amount: {}", [
+      fees.toBigDecimal().toString(),
+      fees.toString(),
+      amount.toString(),
+    ]);
+
+    const inputTokenOne = NetworkConfigs.getTokenAddressFromPoolAddress(
```

Comment:
We should add error checks here to ensure the array has a length of 2, this could cause unwanted runtime errors. I would also `log.error` here.

Same goes for `poolConfig` below

Reply:
Done